src/CrossSigningManager.js

```diff
@@ -70,13 +70,30 @@ async function getSecretStorageKey({ keys: keyInfos }) {
         sdk.getComponent("dialogs.secretstorage.AccessSecretStorageDialog");
     const { finished } = Modal.createTrackedDialog("Access Secret Storage dialog", "",
         AccessSecretStorageDialog,
+        /* props= */
         {
             keyInfo: info,
             checkPrivateKey: async (input) => {
                 const key = await inputToKey(input);
                 return MatrixClientPeg.get().checkSecretStoragePrivateKey(key, info.pubkey);
             },
         },
+        /* className= */ null,
+        /* isPriorityModal= */ false,
+        /* isStaticModal= */ false,
+        /* options= */ {
+            onBeforeClose: async (reason) => {
+                if (reason !== "backgroundClick") {
+                    return true;
+                }
+                const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+                const [sure] = await Modal.createDialog(QuestionDialog, {
```

Comment:
I think we should also set `danger={true}` so that cancel is red.

Reply:
Indeed, I'm doing this in the follow-up PR.